requests_ntlm/requests_ntlm.py

```diff
@@ -86,8 +87,8 @@ def retry_using_http_NTLM_auth(self, auth_header_field, auth_header,
         # get the challenge
         auth_header_value = response2.headers[auth_header_field]
 
-        ntlm_header_value = list(filter(
-            lambda s: s.startswith('NTLM '), auth_header_value.split(',')
+        ntlm_header_value = list(filter(lambda s: s.startswith('NTLM '),
+                                        map(string.lstrip, auth_header_value.split(','))
```

Comment:
So lambda's and map are really slow in python. This would be much better written as

```py
ntlm_header_value = [
   s for s in (val.lstrip() in auth_header_value.split(','))
   if s.startswith('NTLM ')
]
```